### PR TITLE
benchmark: add benchmark for v4.0.0-rc

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

Add benchmark for v4.0.0-rc.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/2858
- Other reference link(s):

### CODE REVIEW

File: benchmark/tpcc-v2.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+---
+title: TiDB TPC-C Performance Test Report -- v4.0 vs. v3.0
+category: benchmark
```
Comment (TomShawn) [inline benchmark/tpcc-v2.md:3]: Please add a summary in this metadata section.

File: benchmark/tpcc-v2.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+# TiDB TPC-C Performance Test Report -- v4.0 vs. v3.0
+
+## Test purpose
+
+This test aims to compare the TPC-C performance of TiDB 4.0 and TiDB 3.0.
+
+## Test version, time, and place
+
+TiDB version: v4.0.0-rc vs. v3.0.8
```
Comment (TomShawn) [inline benchmark/tpcc-v2.md:14]:
```suggestion
TiDB versions: v4.0.0-rc vs. v3.0.8
```

File: benchmark/tpcc-v2.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+allow-concurrent-memtable-write = true
+[server]
+request-batch-enable-cross-command = false
+[pessimistic-txn]
+pipelined = true
+```
+
+### Cluster topology
+
+* 1 TiKV instance is deployed on each `i3.4xlarge`, 3 in total.
```
Comment (TomShawn) [inline benchmark/tpcc-v2.md:103]:
```suggestion
* 1 TiKV instance is deployed on each `i3.4xlarge` machine, 3 in total.
```

File: benchmark/tpcc-v2.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+* 1 TiDB instance is deployed on each `c5.4xlarge`, 2 in total.
```
Comment (TomShawn) [inline benchmark/tpcc-v2.md:104]:
```suggestion
* 1 TiDB instance is deployed on each `c5.4xlarge` machine, 2 in total.
```

File: benchmark/tpcc-v2.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+* 1 PD instance is deployed on each `m5.xlarge`, 3 in total.
```
Comment (TomShawn) [inline benchmark/tpcc-v2.md:105]:
```suggestion
* 1 PD instance is deployed on each `m5.xlarge` machine, 3 in total.
```

File: benchmark/tpcc-v2.md (hunk 1 above)
Comment (ran-huang) [reply benchmark/tpcc-v2.md:3]: Updated. Thanks!

**Comment by TomShawn** [comment]
LGTM
@Little-Wallace PTAL

**Pull request merged**